packages/core/data-transfer/src/strapi/providers/remote-source/index.ts

```diff
@@ -129,8 +129,20 @@ class RemoteStrapiSourceProvider implements ISourceProvider {
     const stream = await this.#createStageReadStream('assets');
     const pass = new PassThrough({ objectMode: true });
 
+    let payloadIndexer = 0;
+    let currentPayloadIndex = 0;
+
     stream
       .on('data', async (payload: Protocol.Client.TransferAssetFlow[]) => {
+        const payloadIndex = payloadIndexer;
```

Comment:
```suggestion
        // TODO: fix the race condition causing multiple streams to open at once so that this check can be removed
        // Keep track of streams and wait to prevent bugs with parallel processing
        const payloadIndex = payloadIndexer;
```